interview-questions.md

```diff
@@ -5,64 +5,56 @@ First, without external resources. Challenge yourself to answer from memory as i
 
 Then, research the question to expand on your answer. Even if you feel you have answered the question completely, there is always something more to learn. Write your researched answer in your OWN WORDS.
 
-INSTRUCTOR EXAMPLE: What is a conditional statement?
-
-  Your answer:
-
-  Researched answer:
-
-
-
 1. What is git? What is the difference between git and Github?
 
-  Your answer:
+  Your answer: I think git is the bridge to transfer files from your computer to the web. And github is like your journal of programs and files of codes you have made.
 
-  Researched answer:
+  Researched answer: Git is a support programmer that helps programmers with speed, data integrity and distribution of programs from one place to another. The difference between git and github is that git is a tool use to manage source code while github is an online server where programmers can connect and upload their work.
 
 
 
 2. Which JavaScript operators will return a Boolean value?
 
-  Your answer:
+  Your answer: typeOf
 
-  Researched answer:
+  Researched answer: || (or sign) and && (and sign) return the Boolean value.
 
 
 
 3. What is an index? What is the difference between index and value?
 
-  Your answer:
+  Your answer: Index is the numbering system of the computer that starts at 0 instead of 1.
 
-  Researched answer:
+  Researched answer: Index is a numberical position in an array. If there are 10 positions inside the array, then the index starts from 0-9.
 
 
 
 4. What is iteration?
 
-  Your answer:
+  Your answer: Iteration is the repetition of something, also called loop. 
 
-  Researched answer:
+  Researched answer: Iteration is the repetition of a process in the programming language.
```

Comment:
This repetition continues until the condition is met or until the amount of times specified is executed.